In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('Dialysis_Facility_Compare_-_Listing_by_Facility.csv')

In [3]:
df.head()

Provider Number  Network                                  Facility Name  \
0            42592       13           DCI - LITTLE ROCK RENAL SERVICES LLC   
1            52761       18                 DAVITA-PREMIER DIALYSIS CENTER   
2            72549        1                  WALLINGFORD DIALYSIS CARE LLC   
3            82524        4                       DSI LAUREL DIALYSIS, LLC   
4            82527        4  FRESENIUS MEDICAL CARE SOUTHERN DELAWARE, LLC   

            Five Star Date Five Star  Five Star Data Availability Code  \
0  01/01/2015 - 12/31/2018         2                                 1   
1  01/01/2015 - 12/31/2018         3                                 1   
2  01/01/2015 - 12/31/2018         5                                 1   
3  01/01/2015 - 12/31/2018         4                                 1   
4  01/01/2015 - 12/31/2018         5                                 1   

                  Address Line 1                 Address Line 2         City  \
0            1910 JOHN BARROW RD                            NaN  LITTLE ROCK   
1           7612 ATLANTIC AVENUE                            NaN       CUDAHY   
2          720 N MAIN STREET EXT                        SUITE 3  WALLINGFORD   
3  LAUREL SQUARE SHOPPING CENTER  30214 SUSSEX HIGHWAY, UNIT #4       LAUREL   
4             9115 ANTIQUE ALLEY                            NaN  BRIDGEVILLE   

  State  ...  95% C.I. (lower limit) for SWR  \
0    AR  ...                             .12   
1    CA  ...                              .2   
2    CT  ...                   Not Available   
3    DE  ...                   Not Available   
4    DE  ...                   Not Available   

  Number of patients in this facility for SWR  \
0                                          56   
1                                         105   
2                                          12   
3                                           8   
4                                           1   

  Standardized First Kidney Transplant Waitlist Ratio PPPW category text  \
0                                                .45         As Expected   
1                                                .53         As Expected   
2                                      Not Available         As Expected   
3                                      Not Available         As Expected   
4                                      Not Available         As Expected   

  Patient prevalent transplant waitlist data availability code  \
0                                                  1             
1                                                  1             
2                                                  1             
3                                                  1             
4                                                  1             

  95% C.I. (upper limit) for PPPW  95% C.I. (lower limit) for PPPW  \
0                            22.8                              3.4   
1                            32.1                             11.8   
2                            74.5                             10.1   
3                            74.2                               .2   
4                            48.4                              2.8   

   Number of patients for PPPW  Percentage of Prevalent Patients Waitlisted  \
0                          100                                          9.2   
1                          209                                         20.1   
2                           24                                         36.4   
3                           19                                          7.4   
4                           44                                         14.2   

                        Location  
0    POINT (-92.372456 34.73624)  
1  POINT (-118.187206 33.966829)  
2   POINT (-72.809521 41.472838)  
3                            NaN  
4   POINT (-75.592475 38.735365)  

[5 rows x 119 columns]

In [4]:
df.tail()

Provider Number  Network                            Facility Name  \
7621           442666        8  DAVITA TENNESSEE VALLEY DIALYSIS CENTER   
7622           522598       11            DAVITA CAPITOL COURT DIALYSIS   
7623           552873       18                  DAVITA BRISTOL DIALYSIS   
7624           682558        7                           FMC - WAUCHULA   
7625           522548       11            DAVITA WEST APPLETON DIALYSIS   

               Five Star Date      Five Star  \
7621  01/01/2015 - 12/31/2018              4   
7622  01/01/2015 - 12/31/2018              3   
7623  01/01/2015 - 12/31/2018  Not Available   
7624  01/01/2015 - 12/31/2018              4   
7625  01/01/2015 - 12/31/2018              3   

      Five Star Data Availability Code                   Address Line 1  \
7621                                 1                  107 WOODLAWN DR   
7622                                 1                   4176 N 56TH ST   
7623                               258                1232 S BRISTOL ST   
7624                                 1                   457 CARLTON ST   
7625                                 1  10130 W APPLETON AVE, SUITE 500   

     Address Line 2          City State  ...  95% C.I. (lower limit) for SWR  \
7621      SUITE 100  JOHNSON CITY    TN  ...                             .01   
7622            NaN     MILWAUKEE    WI  ...                             .05   
7623            NaN     SANTA ANA    CA  ...                   Not Available   
7624            NaN      WAUCHULA    FL  ...                   Not Available   
7625            NaN     MILWAUKEE    WI  ...                             .41   

     Number of patients in this facility for SWR  \
7621                                          39   
7622                                          52   
7623                               Not Available   
7624                                          22   
7625                                          41   

     Standardized First Kidney Transplant Waitlist Ratio PPPW category text  \
7621                                                .25         As Expected   
7622                                                .38         As Expected   
7623                                      Not Available       Not Available   
7624                                      Not Available         As Expected   
7625                                               1.27         As Expected   

     Patient prevalent transplant waitlist data availability code  \
7621                                                  1             
7622                                                  1             
7623                                                258             
7624                                                  1             
7625                                                  1             

     95% C.I. (upper limit) for PPPW  95% C.I. (lower limit) for PPPW  \
7621                            33.1                              2.1   
7622                            32.6                              5.2   
7623                   Not Available                    Not Available   
7624                            35.4                              1.9   
7625                            36.9                              7.1   

      Number of patients for PPPW  \
7621                           76   
7622                           82   
7623                Not Available   
7624                           56   
7625                           92   

      Percentage of Prevalent Patients Waitlisted  \
7621                                          9.5   
7622                                           14   
7623                                Not Available   
7624                                          9.2   
7625                                         17.4   

                           Location  
7621   POINT (-82.398053 36.307557)  
7622   POINT (-87.982914 43.093799)  
7623  POINT (-117.885172 33.731441) 

In [5]:
df.shape

(7626, 119)

In [6]:
for col in df.columns: 
    print(col) 

Provider Number
Network
Facility Name
Five Star Date
Five Star
Five Star Data Availability Code
Address Line 1
Address Line 2
City
State
Zip
County
Phone Number
Profit or Non-Profit
Chain Owned
Chain Organization
Late Shift
# of Dialysis Stations
Offers in-center hemodialysis
Offers peritoneal dialysis
Offers home hemodialysis training
Certification or Recertification Date
Claims Date
CROWNWeb Date
STrR Date
Percentage of Medicare patients with Hgb<10 g/dL
HGB<10 data availability code
Percentage of Medicare patients with Hgb>12 g/dL
Hgb > 12 data availability code
Number of Dialysis Patients with Hgb data
Patient Transfusion data availability Code
Patient Transfusion category text
Number of patients included in the transfusion summary
Percent of Adult HD patients with Kt/V >= 1.2
Adult HD Kt/V data availability code
Percentage of Adult PD PTS with Kt/V >= 1.7
Adult PD Kt/V Data Availability Code
Percentage of Pediatric HD patients with Kt/V >= 1.2
Pediatric HD Kt/V Data Availability C

In [35]:
df.iloc[:, :60].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7626 entries, 0 to 7625
Data columns (total 60 columns):
 #   Column                                                                                   Non-Null Count  Dtype 
---  ------                                                                                   --------------  ----- 
 0   Provider Number                                                                          7626 non-null   int64 
 1   Network                                                                                  7626 non-null   int64 
 2   Facility Name                                                                            7626 non-null   object
 3   Five Star Date                                                                           7626 non-null   object
 4   Five Star                                                                                7626 non-null   object
 5   Five Star Data Availability Code                                     

In [36]:
df.iloc[:, 60:].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7626 entries, 0 to 7625
Data columns (total 59 columns):
 #   Column                                                               Non-Null Count  Dtype 
---  ------                                                               --------------  ----- 
 0   Patient Hospitalization category text                                7626 non-null   object
 1   Patient Hospitalization data availability Code                       7626 non-null   int64 
 2   Patient Hospital Readmission Category                                7626 non-null   object
 3   Patient Hospital Readmission data availability Code                  7626 non-null   int64 
 4   Patient Survival Category Text                                       7626 non-null   object
 5   Patient Survival data availability code                              7626 non-null   int64 
 6   Number of patients included in hospitalization summary               7626 non-null   object
 7   Number of hospi

In [25]:
df.iloc[:, :30].describe()

Provider Number      Network  Five Star Data Availability Code  \
count      7626.000000  7626.000000                       7626.000000   
mean     315876.077105     9.660372                         31.525833   
std      193257.685562     4.708672                         83.403249   
min       12306.000000     1.000000                          1.000000   
25%      142822.250000     6.000000                          1.000000   
50%      322545.500000     9.000000                          1.000000   
75%      452511.750000    14.000000                          1.000000   
max      852568.000000    18.000000                        261.000000   

                Zip   Late Shift  # of Dialysis Stations  \
count   7626.000000  7626.000000             7626.000000   
mean   50297.359691     0.167060               17.463284   
std    27581.675445     0.373054                8.495540   
min      603.000000     0.000000                0.000000   
25%    29533.000000     0.000000               12.000000   
50%    46032.000000     0.000000               17.000000   
75%    76060.750000     0.000000               22.000000   
max    99801.000000     1.000000               80.000000   

       Offers in-center hemodialysis  Offers peritoneal dialysis  \
count                    7626.000000                 7626.000000   
mean                        0.940336                    0.525439   
std                         0.236879                    0.499385   
min                         0.000000                    0.000000   
25%                         1.000000                    0.000000   
50%                         1.000000                    1.000000   
75%                         1.000000                    1.000000   
max                         1.000000                    1.000000   

       Offers home hemodialysis training  HGB<10 data availability code  \
count                         7626.00000                    7626.000000   
mean                             0.29006                      31.839890   
std                              0.45382                      75.349864   
min                              0.00000                       1.000000   
25%                              0.00000                       1.000000   
50%                              0.00000                       1.000000   
75%                              1.00000                       1.000000   
max                              1.00000                     258.000000   

       Hgb > 12 data availability code  
count                      7626.000000  
mean                         31.839890  
std                          75.349864  
min                           1.000000  
25%                           1.000000  
50%                           1.000000  
75%                           1.000000  
max                         258.000000

In [29]:
df.iloc[:, 30:90].describe()

Patient Transfusion data availability Code  \
count                                 7626.000000   
mean                                    41.090742   
std                                     87.672939   
min                                      1.000000   
25%                                      1.000000   
50%                                      1.000000   
75%                                      1.000000   
max                                    258.000000   

       Adult HD Kt/V data availability code  \
count                           7626.000000   
mean                              26.262392   
std                               70.987742   
min                                1.000000   
25%                                1.000000   
50%                                1.000000   
75%                                1.000000   
max                              258.000000   

       Adult PD Kt/V Data Availability Code  \
count                           7626.000000   
mean                             170.362575   
std                              108.505266   
min                                1.000000   
25%                                1.000000   
50%                              201.000000   
75%                              257.000000   
max                              258.000000   

       Pediatric HD Kt/V Data Availability Code  \
count                               7626.000000   
mean                                 255.115132   
std                                   17.913859   
min                                    1.000000   
25%                                  259.000000   
50%                                  259.000000   
75%                                  259.000000   
max                                  259.000000   

       Hypercalcemia Data Availability Code  \
count                           7626.000000   
mean                              13.552190   
std                               52.313945   
min                                1.000000   
25%                                1.000000   
50%                                1.000000   
75%                                1.000000   
max                              258.000000   

       Serum phosphorus Data Availability Code   \
count                                7626.00000   
mean                                   13.12300   
std                                    51.23718   
min                                     1.00000   
25%                                     1.00000   
50%                                     1.00000   
75%                                     1.00000   
max                                   258.00000   

       Patient Hospitalization data availability Code  \
count                                      7626.00000   
mean                                         25.68673   
std                                          74.10966   
min                                           1.00000   
25%                                           1.00000   
50%                                           1.00000   
75%                                           1.00000   
max                                         258.00000   

       Patient Hospital Readmission data availability Code  \
count                                        7626.000000     
mean                                           26.699449     
std                                            74.577879     
min                                             1.000000     
25%                                             1.000000     
50%                                             1.000000     
75%                                             1.000000     
max                                           258.000000     

       Patient Survival data availability code  \
count                              7626.000000   
mean                                 30.239182   
std                                  79.229534   
min                                   1.000000   
25%           

In [30]:
df.iloc[:, 90:].describe()

Fistula data availability code  \
count                     7626.000000   
mean                        22.494099   
std                         67.614074   
min                          1.000000   
25%                          1.000000   
50%                          1.000000   
75%                          1.000000   
max                        258.000000   

       Long term catheter Data Availability Code  nPCR Data Availability Code  \
count                                7626.000000                  7626.000000   
mean                                   21.936271                   254.674928   
std                                    65.879543                    20.264866   
min                                     1.000000                     1.000000   
25%                                     1.000000                   259.000000   
50%                                     1.000000                   259.000000   
75%                                     1.000000                   259.000000   
max                                   258.000000                   259.000000   

       Patient transplant waitlist data availability code  \
count                                        7626.000000    
mean                                           97.691975    
std                                           105.786322    
min                                             1.000000    
25%                                             1.000000    
50%                                             1.000000    
75%                                           199.000000    
max                                           258.000000    

       Patient prevalent transplant waitlist data availability code  
count                                        7626.000000             
mean                                           14.793994             
std                                            55.649087             
min                                             1.000000             
25%                                             1.000000             
50%                                             1.000000             
75%                                             1.000000             
max                                           258.000000

In [9]:
df['Five Star'].unique()

array(['2', '3', '5', '4', 'Not Available', '1'], dtype=object)

In [10]:
df[['Profit or Non-Profit','Percentage of Prevalent Patients Waitlisted']]

Profit or Non-Profit Percentage of Prevalent Patients Waitlisted
0              Non-Profit                                         9.2
1                  Profit                                        20.1
2                  Profit                                        36.4
3                  Profit                                         7.4
4                  Profit                                        14.2
...                   ...                                         ...
7621               Profit                                         9.5
7622               Profit                                          14
7623               Profit                               Not Available
7624               Profit                                         9.2
7625               Profit                                        17.4

[7626 rows x 2 columns]

# Column Names Defined

### Star Rating Explained
The “Dialysis Facility Compare (DFC) Star Program” is a rating system developed by Medicare that assigns 1 to 5 stars to dialysis facilities by comparing the health of the patients in their clinics to the patients in other dialysis facilities across the country.

### Serum phosphorus significance
Serum phosphorus is measured in milligrams of phosphorus per deciliter of blood (mg/dL). According to Mayo Medical Laboratories, a normal range for adults is generally 2.5 to 4.5 mg/dL. The normal range varies slightly depending on your age.
Most commonly, a high level of phosphorus is related to a kidney disorder. It shows that your kidneys are having difficulty clearing phosphorus from your blood. A high level of phosphorus can also mean uncontrolled diabetes and other endocrine disorders.

### Concept of Kt/V
The parameter Kt/V is a measurement of the efficacy of a hemodialysis session. It identifies the effective removal of a specific solute (clearance K) resulting from a given treatment (characterized by time t) in a given patient (with a specific volume of distribution V for the solute considered).

### Hgb g/dL
Hemoglobin is a protein in your red blood cells that carries oxygen from your lungs to the rest of your body. A normal hemoglobin level is 11 to 18 grams per deciliter (g/dL), depending on your age and gender. But 7 to 8 g/dL is a safe level. 

### Hypercalcemia & Serum Calcium
Hypercalcemia is a condition in which the calcium level in your blood is above normal. Too much calcium in your blood can weaken your bones, create kidney stones, and interfere with how your heart and brain work. Hypercalcemia is usually a result of overactive parathyroid glands.
A normal serum calcium level is 8-10 mg/dL (2-2.5 mmol/L) with some interlaboratory variation in the reference range, and hypercalcemia is defined as a serum calcium level greater than 10.5 mg/dL (>2.5 mmol/L).

### AV fistula
The best type of long-term access is an AV fistula. A surgeon connects an artery to a vein, usually in your arm, to create an AV fistula. An artery is a blood vessel that carries blood away from your heart. A vein is a blood vessel that carries blood back toward your heart. When the surgeon connects an artery to a vein, the vein grows wider and thicker, making it easier to place the needles for dialysis. The AV fistula also has a large diameter that allows your blood to flow out and back into your body quickly. The goal is to allow high blood flow so that the largest amount of blood can pass through the dialyzer.

### nPCR
The normalized protein catabolic rate (nPCR) is a formula commonly used to assess dietary protein intake in dialysis patients, as a means towards determining nutritional adequacy, a major problem in many ESRD patients. 
The nPCR is reported in grams of urea nitrogen per kilogram per day.
Most guidelines specify maintaining the protein intake above 1.0 – 1.2 g/kg/day in dialysis patients, with values less than 0.8 g/kg/day being equated with malnutrition.

### Standardized Waitlist Ratio (SWR)
The Standardized Waitlist Ratio (SWR) measure tracks the number of incident patients at the dialysis facility under the age of 75 listed on the kidney or kidney-pancreas transplant waitlist or who received a living donor transplant within the first year of initiating dialysis. SWR is calculated to compare the observed waitlisting rate in the facility to the waitlisting rate that was expected. 

# Preliminary thoughts for data analysis
I'm interested in comparing dialysis star rating to patient health indicators (nPCR, serum calcium, Hgb, Kt/V, serum phosphorus).
Also might be interesting to compare patient health indicators to a center's profit/ non-profit status or whether or not it's a chain buisness. 
Another idea is the relationship between star rating or profit/ non-profit status vs SWR.